### ISSUE TITLE
SIGHASH_UTXOS will need to take into account interaction with Double-Spend proofs

### ISSUE THREAD

**Issue by cculianu** [bitjson/cashtokens#55]
The current implementation of DSProofs will not be able to encapsulate the hash of the UTXOs the proof.  As such, DSProofs will not be possible to protect and/or provide a proof for an input that is spent using `SIGHASH_UTXOS`.

This caveat needs to be mentioned in the spec.  Ultimately, the DSProof spec needs to be expanded to cover the additional hash preimage data that must be shared as part of a DSProof. (Namely, the hash of the utxos).

**Comment by bitjson** [comment]
Thanks for opening an issue! Would you mind making a PR for this when you get a chance?

**Comment by cculianu** [comment]
Yes, I will definitely do so ASAP.  We basically need to extend DSP to cover `SIGHASH_UTXO` 's `hashUtxos` ... so probably the spec just needs to discuss that caveat and recommend a DSProof spec update.. But yes. Will do so. 👍
